src/Process/3dTilesProcessing.js

```diff
@@ -0,0 +1,174 @@
+import * as THREE from 'three';
+import Fetcher from '../Core/Scheduler/Providers/Fetcher';
+import { $3dTilesIndex } from '../Core/Scheduler/Providers/3dTiles_Provider';
+
+function requestNewTile(view, scheduler, geometryLayer, metadata, parent) {
+    const command = {
+        /* mandatory */
+        view,
+        requester: parent,
+        layer: geometryLayer,
+        priority: 10000,
+        /* specific params */
+        metadata,
+        redraw: false,
+    };
+
+    return scheduler.execute(command);
+}
+
+function subdivideNode(context, layer, node) {
+    if (!node.pendingSubdivision && node.children.filter(n => n.layer == layer.id).length == 0) {
+        node.pendingSubdivision = true;
+
+        const childrenTiles = layer.tileIndex.index[node.tileId].children;
+        if (childrenTiles === undefined) {
+            return;
+        }
+
+        const promises = [];
+        for (let i = 0; i < childrenTiles.length; i++) {
+            promises.push(
+                requestNewTile(context.view, context.scheduler, layer, childrenTiles[i], node).then((tile) => {
+                    node.add(tile);
+                    tile.updateMatrixWorld();
+                    if (node.additiveRefinement) {
+                        context.view.notifyChange(0, true);
+                    }
+                }));
+        }
+
+        Promise.all(promises).then(() => {
+            node.pendingSubdivision = false;
+            context.view.notifyChange(0, true);
+        });
+    }
+}
+
+export function $3dTilesCulling(node, camera) {
+    if (node.boundingVolume.region) {
+        // TODO
+    }
+    if (node.boundingVolume.box) {
+        return !camera.isBox3DVisible(node.boundingVolume.box, node.matrixWorld);
+    } else if (node.boundingVolume.sphere) {
+        // TODO return !camera.isSphereVisible(node.boundingVolume.sphere, node.matrixWorld);
+    }
+    return false;
+}
+
+let preSSE;
+export function pre3dTilesUpdate(context) {
+    // pre-sse
+    const hypotenuse = Math.sqrt(context.camera.width * context.camera.width + context.camera.height * context.camera.height);
+    const radAngle = context.camera.FOV * Math.PI / 180;
+
+     // TODO: not correct -> see new preSSE
+    // const HFOV = 2.0 * Math.atan(Math.tan(radAngle * 0.5) / context.camera.ratio);
+    const HYFOV = 2.0 * Math.atan(Math.tan(radAngle * 0.5) * hypotenuse / context.camera.width);
```

Comment:
From the [three.js doc](https://threejs.org/docs/index.html#api/cameras/PerspectiveCamera) I understand it `context.camera.FOV` (`radAngle`) is the vertical field of view. So I wonder why `HYFOV` is not calculated in terms of `camera.height` instead of `camera.width`:

    const HYFOV = 2.0 * Math.atan(Math.tan(radAngle * 0.5) * hypotenuse / context.camera.height);

Also, I wonder why you use the hypotenuse FOV for the SSE calculation. Why not using the vertical FOV as in Cesium? Just curious…